torch/nn/functional.py

```diff
@@ -1328,13 +1345,27 @@ def dropout3d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
         training: apply dropout if is ``True``. Default: ``True``
         inplace: If set to ``True``, will do this operation in-place. Default: ``False``
     """
-    # This is 100% the same code as dropout2d. We duplicate this code so that
-    # stack traces are not confusing.
     if has_torch_function_unary(input):
         return handle_torch_function(dropout3d, (input,), input, p=p, training=training, inplace=inplace)
     if p < 0.0 or p > 1.0:
         raise ValueError("dropout probability has to be between 0 and 1, " "but got {}".format(p))
-    return _VF.feature_dropout_(input, p, training) if inplace else _VF.feature_dropout(input, p, training)
+    inp_dim = input.dim()
+    msg = f"dropout3d: Input should be 4-D (unbatched) or 5-D (batched) but received {inp_dim}-D tensor"
+    assert inp_dim in (3, 4, 5), msg
+    if inp_dim == 3:
+        warn_msg = ("dropout3d: 3-D input is being deprecated for Dropout3d."
+                    " Look at the docs for correct input shapes and usage.")
```

Comment:
nit: deprecation should be actionable to remove the warning - in this case, switching to `F.dropout` is the way to go:
```
Received a 3D input to dropout3d, which is deprecated and will result in an error in a future release. To retain the behavior and silence this warning, please use dropout instead. Note that dropout3d exists to provide channel-wise dropout on inputs with 3 spatial dimensions, a channel dimension, and an optional batch dimension (i.e. 4D or 5D inputs).
```